connector/src/main/java/org/geysermc/connector/network/translators/world/block/entity/SkullBlockEntityTranslator.java

```diff
@@ -68,4 +88,119 @@ public CompoundTag getDefaultBedrockTag(String bedrockId, int x, int y, int z) {
+                    break;
+                case 6031: //South
+                    rotation = 0;
+                    z -= 0.24f;
+                    break;
+                case 6032: //West
+                    rotation = 90;
+                    x += 0.24f;
+                    break;
+                case 6033: //East
+                    rotation = 270;
+                    x -= 0.24f;
+                    break;
+            }
+        } else {
+            rotation = (180f + (BlockStateValues.getSkullRotation(blockState) * 22.5f)) % 360;
+        }
+
+        UUID uuid = UUID.randomUUID();
+        long geyserId = session.getEntityCache().getNextEntityId().incrementAndGet();
+
+        PlayerEntity player = new PlayerEntity(new GameProfile(uuid, ""), 1, geyserId, Vector3f.from(x), Vector3f.from(y), Vector3f.from(z));
+
+        //Creates a fake player to be the custom skull
+        PlayerSkinPacket playerSkinPacket = new PlayerSkinPacket();
+        playerSkinPacket.setOldSkinName("OldName");
+        playerSkinPacket.setNewSkinName("NewName");
+        playerSkinPacket.setSkin(skin);
+        playerSkinPacket.setUuid(uuid);
+
+        //Set bounding box to almost nothing so the skull is able to be broken and not cause entity to cast a shadow
+        EntityDataMap metadata = new EntityDataMap();
+        metadata.put(EntityData.SCALE, 1.08f);
+        metadata.put(EntityData.BOUNDING_BOX_HEIGHT, 0.001f);
+        metadata.put(EntityData.BOUNDING_BOX_WIDTH, 0.001f);
+        metadata.getOrCreateFlags().setFlag(EntityFlag.INVISIBLE, true); //Set invisible so you don't see Steve while the model changes to the custom skull
+
+        player.setMetadata(metadata);
+        player.spawnEntity(session);
+
+        //Required for the fake player to show up at all
+        AddPlayerPacket addPlayerPacket = new AddPlayerPacket();
+        addPlayerPacket.setUuid(uuid);
+        addPlayerPacket.setUsername("");
+        addPlayerPacket.setRuntimeEntityId(geyserId);
+        addPlayerPacket.setUniqueEntityId(geyserId);
+        addPlayerPacket.setPosition(Vector3f.from(x, y, z));
+        addPlayerPacket.setRotation(Vector3f.from(0f, 0f, 0f));
+        addPlayerPacket.getMetadata().putAll(metadata);
+
+        //Send all the packets back to player
+        session.sendUpstreamPacket(playerSkinPacket);
+        session.sendUpstreamPacket(addPlayerPacket);
+
+        player.updateBedrockAttributes(session);
+        player.moveAbsolute(session, Vector3f.from(x, y, z), rotation, 0, true, false);
+        CACHED_SKULLS.put((new Position((int) tag.get("x").getValue(), (int) tag.get("y").getValue(), (int) tag.get("z").getValue())), player);
+        session.getConnector().getGeneralThreadPool().schedule(() -> {
+            metadata.getFlags().setFlag(EntityFlag.INVISIBLE, false);
+            player.updateBedrockMetadata(session);
+        }, 500, TimeUnit.MILLISECONDS); //Delay 5 seconds to give the model time to load in

```

Comment:
Comment should read `// Delay 500 milliseconds to give the model time to load in`